src/main/java/org/fcrepo/migration/Migrator.java

```diff
@@ -129,19 +137,44 @@ public Migrator(final ObjectSource source, final StreamingFedoraObjectHandler ha
 
     /**
      * the run method for migrator.
+     *
      * @throws XMLStreamException xml stream exception
      */
     public void run() throws XMLStreamException {
         int index = 0;
-        for (final FedoraObjectProcessor o : source) {
-            final String pid = o.getObjectInfo().getPid();
-            if (pid != null) {
-
-                // Process if limit is '-1', or we have not hit the non-negative 'limit'...
-                //  ..and the PidListManager accepts the PID
-                if ((limit < 0 || index++ < limit) && acceptPid(pid)) {
-                    LOGGER.info("Processing \"" + pid + "\"...");
-                    o.processObject(handler);
+
+        final var iterator = source.iterator();
+        while (true) {
```

Comment:
Is there a reason to use a forever loop here instead of:

```java
for (final var iterator = source.iterator(); iterator.hasNext();) {
    final var o = iterator.next();
}
``

Reply:
No reason - I will change.